tests/framework/filters/HttpCacheTest.php

```
@@ -48,6 +48,25 @@ public function testEmptyPragma(): void
+            return 'foo';
+        };
+
+        $this->assertTrue($httpCache->beforeAction(null));
+
+        $etag = Yii::$app->getResponse()->getHeaders()->get('Etag');
+        $this->assertNotNull($etag);
+
+        Yii::$app->getRequest()->headers->set('If-None-Match', $etag);
+        $this->assertFalse($httpCache->beforeAction(null));
+        $this->assertSame(304, Yii::$app->getResponse()->getStatusCode());
+    }
```

PHPMD (2.15.0)

[error] testQueryMethodIsCached accesses the super-global variable $_SERVER. (undefined)

(Superglobals)